<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>AlphaMattingWidget</class>
 <widget class="QWidget" name="AlphaMattingWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>248</width>
    <height>522</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Alpha Matting</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_4">
   <item>
    <widget class="QGroupBox" name="mattingGroupBox">
     <property name="maximumSize">
      <size>
       <width>16777215</width>
       <height>160</height>
      </size>
     </property>
     <property name="title">
      <string>Mattting</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_3">
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <widget class="QRadioButton" name="inputImageRadioButton">
          <property name="text">
           <string>Input Image</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QToolButton" name="inputImgToolButton">
          <property name="text">
           <string>...</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <widget class="QRadioButton" name="trimapImageRadioButton">
          <property name="text">
           <string>Trimap Image</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QToolButton" name="trimapImgToolButton">
          <property name="text">
           <string>...</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <widget class="QRadioButton" name="outputAlphaRadioButton">
          <property name="text">
           <string>Output Alpha</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="outputColorRadioButton">
          <property name="text">
           <string>Output Color</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_5">
        <item>
         <widget class="QPushButton" name="runPushButton">
          <property name="text">
           <string>Run</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="savePushButton">
          <property name="text">
           <string>Save</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="filteringGroupBox">
     <property name="minimumSize">
      <size>
       <width>0</width>
       <height>279</height>
      </size>
     </property>
     <property name="title">
      <string>Smoothing</string>
     </property>
     <property name="checkable">
      <bool>true</bool>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <widget class="QGroupBox" name="bilateralGroupBox">
        <property name="title">
         <string>Bilateral Filter</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
        <property name="checked">
         <bool>false</bool>
        </property>
        <layout class="QGridLayout" name="gridLayout">
         <item row="0" column="0">
          <widget class="QLabel" name="bilateralKernelLabel">
           <property name="text">
            <string>Kernel Size</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QSpinBox" name="bilateralKernelSpinBox">
           <property name="minimum">
            <number>3</number>
           </property>
           <property name="maximum">
            <number>99</number>
           </property>
           <property name="singleStep">
            <number>2</number>
           </property>
           <property name="value">
            <number>3</number>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="bilateralSpatialLabel">
           <property name="text">
            <string>Spatial Radius</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QDoubleSpinBox" name="bilateralSpatialSpinBox">
           <property name="singleStep">
            <double>1.000000000000000</double>
           </property>
           <property name="value">
            <double>3.000000000000000</double>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="bilateralIntensityLabel">
           <property name="text">
            <string>Intensity Radius</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QDoubleSpinBox" name="bilateralIntensitySpinBox">
           <property name="singleStep">
            <double>1.000000000000000</double>
           </property>
           <property name="value">
            <double>3.000000000000000</double>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="guidedGroupBox">
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>135</height>
         </size>
        </property>
        <property name="title">
         <string>Guided Filter</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
        <layout class="QGridLayout" name="gridLayout_2">
         <item row="1" column="0">
          <widget class="QLabel" name="guidedKernelLabel">
           <property name="text">
            <string>Radius</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QSpinBox" name="guidedKernelSpinBox">
           <property name="minimum">
            <number>1</number>
           </property>
           <property name="maximum">
            <number>99</number>
           </property>
           <property name="singleStep">
            <number>1</number>
           </property>
           <property name="value">
            <number>4</number>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="guidedRegularizationLabel">
           <property name="text">
            <string>Regularization</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QDoubleSpinBox" name="guidedRegularizationSpinBox">
           <property name="decimals">
            <number>5</number>
           </property>
           <property name="singleStep">
            <double>0.001000000000000</double>
           </property>
           <property name="value">
            <double>0.000010000000000</double>
           </property>
          </widget>
         </item>
         <item row="0" column="0" colspan="2">
          <widget class="QGroupBox" name="guidedImageGroupBox">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>50</height>
            </size>
           </property>
           <property name="title">
            <string>Guide Image</string>
           </property>
           <property name="checkable">
            <bool>false</bool>
           </property>
           <property name="checked">
            <bool>false</bool>
           </property>
           <layout class="QHBoxLayout" name="horizontalLayout_3">
            <item>
             <widget class="QRadioButton" name="guideInputColorRadioButton">
              <property name="text">
               <string>Input Color</string>
              </property>
              <property name="checked">
               <bool>false</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QRadioButton" name="guideOutputAlphaRadioButton">
              <property name="text">
               <string>Output Alpha</string>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="backgroundImageGroupBox">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="minimumSize">
      <size>
       <width>0</width>
       <height>53</height>
      </size>
     </property>
     <property name="title">
      <string>Background Image</string>
     </property>
     <property name="checkable">
      <bool>true</bool>
     </property>
     <property name="checked">
      <bool>false</bool>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_4">
      <item>
       <widget class="QLineEdit" name="backgroundImageLineEdit"/>
      </item>
      <item>
       <widget class="QToolButton" name="backgroundImageToolButton">
        <property name="text">
         <string>...</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>inputImgToolButton</sender>
   <signal>clicked()</signal>
   <receiver>AlphaMattingWidget</receiver>
   <slot>openInputImage()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>233</x>
     <y>42</y>
    </hint>
    <hint type="destinationlabel">
     <x>132</x>
     <y>249</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>trimapImgToolButton</sender>
   <signal>clicked()</signal>
   <receiver>AlphaMattingWidget</receiver>
   <slot>openTrimapImage()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>233</x>
     <y>69</y>
    </hint>
    <hint type="destinationlabel">
     <x>132</x>
     <y>249</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>inputImageRadioButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>AlphaMattingWidget</receiver>
   <slot>onInputRadioButtonToggled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>117</x>
     <y>42</y>
    </hint>
    <hint type="destinationlabel">
     <x>132</x>
     <y>249</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>trimapImageRadioButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>AlphaMattingWidget</receiver>
   <slot>onTrimapRadioButtonToggled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>117</x>
     <y>69</y>
    </hint>
    <hint type="destinationlabel">
     <x>132</x>
     <y>249</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>outputAlphaRadioButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>AlphaMattingWidget</receiver>
   <slot>onOutputAlphaRadioButtonToggled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>132</x>
     <y>94</y>
    </hint>
    <hint type="destinationlabel">
     <x>132</x>
     <y>249</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>savePushButton</sender>
   <signal>clicked()</signal>
   <receiver>AlphaMattingWidget</receiver>
   <slot>onSaveButtonClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>218</x>
     <y>482</y>
    </hint>
    <hint type="destinationlabel">
     <x>132</x>
     <y>251</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>runPushButton</sender>
   <signal>clicked()</signal>
   <receiver>AlphaMattingWidget</receiver>
   <slot>onRunButtonClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>106</x>
     <y>123</y>
    </hint>
    <hint type="destinationlabel">
     <x>132</x>
     <y>324</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>outputColorRadioButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>AlphaMattingWidget</receiver>
   <slot>onOutputColorRadioButtonToggled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>122</x>
     <y>118</y>
    </hint>
    <hint type="destinationlabel">
     <x>122</x>
     <y>250</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>filteringGroupBox</sender>
   <signal>toggled(bool)</signal>
   <receiver>AlphaMattingWidget</receiver>
   <slot>onFilteringToggled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>122</x>
     <y>304</y>
    </hint>
    <hint type="destinationlabel">
     <x>122</x>
     <y>250</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>bilateralGroupBox</sender>
   <signal>toggled(bool)</signal>
   <receiver>AlphaMattingWidget</receiver>
   <slot>onBilateralFilterToggled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>122</x>
     <y>250</y>
    </hint>
    <hint type="destinationlabel">
     <x>122</x>
     <y>250</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>guidedGroupBox</sender>
   <signal>toggled(bool)</signal>
   <receiver>AlphaMattingWidget</receiver>
   <slot>onGuidedFilterToggled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>122</x>
     <y>366</y>
    </hint>
    <hint type="destinationlabel">
     <x>122</x>
     <y>250</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>bilateralKernelSpinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>AlphaMattingWidget</receiver>
   <slot>onBilateralFilterUpdate()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>170</x>
     <y>230</y>
    </hint>
    <hint type="destinationlabel">
     <x>122</x>
     <y>250</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>bilateralSpatialSpinBox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>AlphaMattingWidget</receiver>
   <slot>onBilateralFilterUpdate()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>170</x>
     <y>256</y>
    </hint>
    <hint type="destinationlabel">
     <x>122</x>
     <y>250</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>bilateralIntensitySpinBox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>AlphaMattingWidget</receiver>
   <slot>onBilateralFilterUpdate()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>170</x>
     <y>282</y>
    </hint>
    <hint type="destinationlabel">
     <x>122</x>
     <y>250</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>guidedKernelSpinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>AlphaMattingWidget</receiver>
   <slot>onGuidedFilterUpdate()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>170</x>
     <y>377</y>
    </hint>
    <hint type="destinationlabel">
     <x>122</x>
     <y>250</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>guidedRegularizationSpinBox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>AlphaMattingWidget</receiver>
   <slot>onGuidedFilterUpdate()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>170</x>
     <y>403</y>
    </hint>
    <hint type="destinationlabel">
     <x>122</x>
     <y>250</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>guideInputColorRadioButton</sender>
   <signal>clicked()</signal>
   <receiver>AlphaMattingWidget</receiver>
   <slot>onGuidedFilterUpdate()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>77</x>
     <y>364</y>
    </hint>
    <hint type="destinationlabel">
     <x>122</x>
     <y>262</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>guideOutputAlphaRadioButton</sender>
   <signal>clicked()</signal>
   <receiver>AlphaMattingWidget</receiver>
   <slot>onGuidedFilterUpdate()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>163</x>
     <y>364</y>
    </hint>
    <hint type="destinationlabel">
     <x>122</x>
     <y>262</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>guidedImageGroupBox</sender>
   <signal>toggled(bool)</signal>
   <receiver>AlphaMattingWidget</receiver>
   <slot>onGuidedFilterUpdate()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>122</x>
     <y>358</y>
    </hint>
    <hint type="destinationlabel">
     <x>122</x>
     <y>262</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <signal>showImage(QImage)</signal>
  <signal>filterChanged(int)</signal>
  <signal>bilateralFilterChanged(int,float,float)</signal>
  <signal>guidedFilterChanged(int,float,bool,QImage)</signal>
  <slot>openInputImage()</slot>
  <slot>openTrimapImage()</slot>
  <slot>onInputRadioButtonToggled(bool)</slot>
  <slot>onTrimapRadioButtonToggled(bool)</slot>
  <slot>onOutputAlphaRadioButtonToggled(bool)</slot>
  <slot>onOutputColorRadioButtonToggled(bool)</slot>
  <slot>onSaveButtonClicked()</slot>
  <slot>onRunButtonClicked()</slot>
  <slot>onFilteringToggled(bool)</slot>
  <slot>onBilateralFilterToggled(bool)</slot>
  <slot>onGuidedFilterToggled(bool)</slot>
  <slot>onBilateralFilterUpdate()</slot>
  <slot>onGuidedFilterUpdate()</slot>
 </slots>
</ui>
